pymeasure/instruments/yokogawa/yokogawags200.py

```diff
@@ -0,0 +1,147 @@
+        validator=strict_discrete_set,
+        values={'current': 'CURR', 'voltage': 'VOLT'},
+        get_process=lambda s: s.strip()
+    )
+
+    source_range = Instrument.control(
+        ":SOURce:RANGe?",
+        "SOURce:RANGe %g",
+        """Floating point number that controls the range (either in voltage or current) 
+        of the output. "Range" refers to the maximum source level. """,
+        validator=truncated_discrete_set,
+        values=[1e-3, 10e-3, 100e-3, 200e-3, 1, 10, 30]
+    )
+
+    voltage_limit = Instrument.control(
+        "SOURce:PROTection:VOLTage?",
+        "SOURce:PROTection:VOLTage %g",
+        """Floating point number that controls the voltage limit. "Limit" refers to maximum 
+        value of the electrical value that is conjugate to the mode (current is conjugate to 
+        voltage, and vice versa). Thus, voltage limit is only applicable when in 'current' mode""",
+        validator=truncated_range,
+        values=[1, 30]
+    )
+
+    current_limit = Instrument.control(
+        "SOURce:PROTection:CURRent?",
+        "SOURce:PROTection:CURRent %g",
+        """Floating point number that controls the current limit. "Limit" refers to maximum value 
+        of the electrical value that is conjugate to the mode (current is conjugate to voltage, 
+        and vice versa). Thus, current limit is only applicable when in 'voltage' mode""",
+        validator=truncated_range,
+        values=[1e-3, 200e-3]
+    )
+
+    def __init__(self, adapter, **kwargs):
+        super(YokogawaGS200, self).__init__(
+            adapter, "Yokogawa GS200 Source", **kwargs
+        )
+
+    @property
+    def source_level(self):
+        """ Floating point number that controls the output level, either a voltage or a current,
+        depending on the source mode.
+        """
+        return float(self.ask(":SOURce:LEVel?"))
+
+    @source_level.setter
+    def source_level(self, level):
+        if level > self.source_range * 1.2:
+            raise ValueError(
+                "Level must be within 1.2 * source_range, otherwise the Yokogawa will produce an "
+                "error."
+            )
+        else:
+            self.write("SOURce:LEVel %g" % level)
+
+    def trigger_ramp_to_level(self, level, ramp_time):
+        """
+        Ramp the output level from its current value to "level" in time "ramp_time". This method
+        will NOT wait until the ramp is finished (thus, it will not block further code evaluation).
+
+        :param float level: final output level
+        :param float ramp_time: time in seconds to ramp
+        :return: None
+        """
+        if not self.source_enabled:
+            raise ValueError(
+                "YokogawaGS200 source must be enabled in order to ramp to a specified level. "
+                "Otherwise, the Yokogawa will reject the ramp."
+            )
+        if ramp_time < MIN_RAMP_TIME:
+            warnings.warn(
```

Comment:
Instead of the `warn`, please use a logger. At the top of the file (after imports), add
```
log = logging.getLogger(__name__)
log.addHandler(logging.NullHandler())
```

then here, use `log.warning(...` instead of `warnings.warn(`

Reply:
this reflect usage in other instruments.